pkg/controller/atlasproject/teams.go

```diff
@@ -194,12 +194,16 @@ func (r *AtlasProjectReconciler) updateTeamState(ctx *workflow.Context, project
 	teamCtx.Client = atlasClient
 
 	if len(assignedProjects) == 0 {
-		log.Debugf("team %s has no project associated to it. removing from atlas.", team.Spec.Name)
-		_, err = teamCtx.Client.Teams.RemoveTeamFromOrganization(ctx.Context, orgID, team.Status.ID)
-		if err != nil {
-			return err
+		if r.ObjectDeletionProtection {
```

Comment:
You also need to check for the annotation `mongodb.com/atlas-resource-policy`

Reply:
Thanks for the heads up! Will fix

Reply:
Done